card/src/test/java/com/adyen/checkout/card/DefaultCardDelegateTest.kt

```diff
@@ -184,6 +184,65 @@ internal class DefaultCardDelegateTest(
             }
         }
 
+        @Test
+        fun `When the a card brand is detected, isCardListVisible should be false`() = runTest {
+            val supportedCardTypes = listOf(CardType.VISA)
+            delegate = createCardDelegate(
+                configuration = getDefaultCardConfigurationBuilder()
+                    .setSupportedCardTypes(*supportedCardTypes.toTypedArray())
+                    .build()
+            )
+            delegate.initialize(CoroutineScope(UnconfinedTestDispatcher()))
+
+            delegate.outputDataFlow.test {
+                delegate.updateComponentState(createOutputData())
+                delegate.updateInputData { /* Empty to trigger an update */ }
+                with(expectMostRecentItem()) {
+                    assertFalse(isCardListVisible)
+                }
+            }
+        }
+
+        @Test
+        fun `When the a card brand is not detected, isCardListVisible should be true`() = runTest {
+            val supportedCardTypes = listOf(CardType.VISA)
+            delegate = createCardDelegate(
+                configuration = getDefaultCardConfigurationBuilder()
+                    .setSupportedCardTypes(*supportedCardTypes.toTypedArray())
+                    .build()
+            )
+            delegate.initialize(CoroutineScope(UnconfinedTestDispatcher()))
+            detectCardTypeRepository.detectionResult = TestDetectCardTypeRepository.TestDetectedCardType.ERROR
```

Comment:
Maybe it's better to add a new `EMPTY` entry to `TestDetectCardTypeRepository.TestDetectedCardType` that emits an empty list, because with `ERROR` nothing will be emitted, instead of an empty list